packages/assets-controllers/src/NftController.ts

```diff
@@ -468,7 +470,7 @@ export class NftController extends BaseController<
   }
 
   getNftApi() {
-    // TODO: Either fix this lint violation or explain why it's necessary to ignore.
+    // False negative.
     // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
     return `${NFT_API_BASE_URL}/tokens`;
```

Comment:
would `${NFT_API_BASE_URL as string}/tokens` fix the warning?

Reply:
[Checking this now](https://github.com/MetaMask/core/pull/4506/commits/e14f03f46127023c4c62e0d367b8b7fc8a8559af), funny part is that it doesn't fail locally, only in CI 😅

Reply:
We probably want to opt for the `eslint-disable` directive instead of a type assertion here, since the cause of this lint error is abnormal behavior from the lint rule, not the code itself. 

The type assertion would silence other potential errors with `NFT_API_BASE_URL` (e.g. if it were to be typed as implicit `any` due to an upstream import issue, the `as string` would suppress useful alarms).

Reply:
I agree that assertion here will silence the potential errors but before that we most probably would have some failed unit tests at first place. Since this is not a complex cast I don't think this makes much sense. 

I am reverting this change and leave this as is.

Reply:
@MajorLift do you have any idea why this is not failing locally but on CI?

Reply:
Yes, I experienced that as well. `yarn lint:eslint` results were different on local as well so it wasn't just an issue with my IDE's eslint server. 

I do remember that it turned out to be an issue with my local configuration, but I can't recall what solved it just now. It definitely wasn't outside of the range of usual-suspect fixes (removing eslint cache, reinstalling packages etc.). I'll leave a comment here if I can remember what it was!